src/Components/Common/AzureComponent.cs

```diff
@@ -53,7 +53,15 @@ internal void AddClient(
         var configSection = builder.Configuration.GetSection(configurationSectionName);
 
         var settings = new TSettings();
+        // Bind both top-level and named configuration sections to the settings object
+        // to allow connection-specific settings.
         BindSettingsToConfiguration(settings, configSection);
+        BindSettingsToConfiguration(settings, configSection.GetSection(connectionName));
+        // Support service key-based binding for clients that support it (e.g. WebPubSubServiceClient).
+        if (configSection.GetSection($"{connectionName}:{serviceKey}").Exists())
+        {
+            BindSettingsToConfiguration(settings, configSection.GetSection($"{connectionName}:{serviceKey}"));
+        }
 
         Debug.Assert(settings is IConnectionStringSettings, $"The settings object should implement {nameof(IConnectionStringSettings)}.");
         if (settings is IConnectionStringSettings csSettings &&
```

Comment:
Sorry I missed this earlier. But on line 96 we also need to bind the `ClientOptions` to both sections.

```
BindClientOptionsToConfiguration(clientBuilder, configSection.GetSection("ClientOptions"));
```

The current code is now breaking this for "Keyed" services because it is only binding ClientOptions to the root section now, and not the keyed named section. (I'm surprised we don't have a test that catches this.)

Reply:
We didn't have any tests here. I suspect it's because it's fairly non-trivial to add generic tests that validate the `TClientOption` type for each resource was configured correctly? I added coverage for ServiceBus in https://github.com/dotnet/aspire/pull/5827/commits/fe0b8893bff3316d14277fdbb7d4ebbfa0179d03 since it propagates the `Identifier` value from options into the top-level client.